New Feature: Notes for unit options

TheFakeCake opened the pull request:
### What
This PR adds a new feature to make visible text notes under unit options (equipment, armour and options).

It also comes with a thorough check, french translation and port to the new options notes of the Orc & Goblin Tribes and Dwarfen Mountain Holds.

Since it adds new data in army JSON files, it shouldn't be conflicting with anything.

### Why
This is useful to give additional information, like `0-1 per army` for an option or `Scaly skin` for an armour.

At the moment, some options have additional text directly in their name, but then it also appears in the unit profile, which is not ideal.

Example:
![option-with-info-in-name](https://github.com/user-attachments/assets/4122a09d-f0cc-4fa2-ad31-607e0788e837)

Also, we can make it visually more pleasing 😅

### How
By adding a `"notes"` object or array of object in a unit equipment, armor or option (in army JSON files).
For example, for the Orc Boar Chariot options, it looks like this: 
```javascript
"options": [
  {
    "name_en": "Third Orc crew member",
    "name_de": "Third Orc crew member",
    "name_fr": "Troisième Aurige Orque",
    "points": 5
  },
  {
    "name_en": "Frenzy",
    "name_de": "Frenzy",
    "name_fr": "Frénésie",
    "points": 4,
    "notes": [    // <---- HERE
      {
        "name_en": "If 2 crew members",
        "name_fr": "Si 2 auriges"
      },
      {
        "name_en": "0-1 per 1000 points",
        "name_fr": "0-1 par tranche de 1000 points"
      }
    ]
  },
  {
    "name_en": "Frenzy",
    "name_de": "Frenzy",
    "name_fr": "Frénésie",
    "points": 6,
    "notes": [    // <---- HERE
      {
        "name_en": "If 3 crew members",
        "name_fr": "Si 3 auriges"
      },
      {
        "name_en": "0-1 per 1000 points",
        "name_fr": "0-1 par tranche de 1000 points"
      }
    ]
  },
  {
    "name_en": "Warpaint",
    "name_de": "Warpaint",
    "name_fr": "Peintures de Guerre",
    "points": 10,
    "notes": {    // <---- HERE
      "name_en": "If frenzied",
      "name_fr": "Si frénétique"
    }
  }
],
```
And the result:
![option-with-new-notes](https://github.com/user-attachments/assets/df969e9e-484e-4cbf-bc10-1b1f6eee2d36)

TheFakeCake commented:
I just saw that @nthiebes is working on a branch with a validation feature for list rules. This PR may not be relevant then.

Should I make another one containing only the fixes and french translations or would it be to much work for you ?

Maybe I can base it on your working branch with validation ?

nthiebes commented on src/components/rules-index/rules-map.js:
```diff
@@ -57,6 +57,7 @@ const additionalOWBRules = {
   "troll magic": {
     url: "the-lores-of-magic/lore-of-troll-magic",
   },
+  "da boss's trophy rack": { url: "magic-item/da-boss-trophy-rack" },
```
Seems to be a typo on whfb.app, I'll let the dev know so we don't need a fix on our side 👍

nthiebes replied:
Done. Will be fixed soon, so you can remove your fix.

nthiebes commented on src/utils/unit.js:
```diff
@@ -262,3 +262,13 @@ export const getUnitName = ({ unit, language }) => {
+export const showUnitOptionNotes = (notes, keyPrefix, className, language) => {
+  return (Array.isArray(notes) ? [...notes] : notes ? [notes] : []).map(
+    (note, index) => (
+      <div className={className} key={`${keyPrefix}-${index}`}>
```
You can make it a paragraph.

nthiebes reviewed (commented):
Nice feature, thanks! 🎉 I'll test it tomorrow but looks good so far 👍.

Maybe it's a good idea to separate the translations. They are changing a lot right now and only cause conflicts with this branch.

nthiebes commented on src/pages/unit/Unit.js:
```diff
@@ -813,33 +821,42 @@ export const Unit = ({ isMobile, previewData = {} }) => {
-                      </i>
-                    </label>
-                  </div>
+                  <>
```
The key would need to move to the fragment.

TheFakeCake commented:
I updated the PR :
- Removed the "da boss's trophy rack" fix
- Switched to <p> for option notes
- Fixed the missing keys on fragments
- Fixed conflicts

nthiebes commented:
Awesome! I tested the new feature and it works great 👍.

Only the translation changes I'd like to exclude in this PR so we can merge it. Some translations (I can only judge German) sound like google translations).

TheFakeCake commented:
I'll do it tomorrow and keep only the one commit with changes to `Unit.js` etc, removing everything else from the PR.

I guess I'll do another PR to add option notes, fixes and translations

TheFakeCake commented:
There it is, this PR now only adds the functionality for option notes.

I'll make other PRs to add translations and option notes.
